scripts/common.sh

```diff
@@ -216,6 +216,68 @@ gcloud_auth() {
     gcloud auth login --update-adc
 }
 
+ensure_tf_backend() {
+  GCP_CREDENTIALS_PATH="${1}"
+  shift
+  TF_STATA_BUCKET_NAME="gs://tf-state-${DEFAULT_PROJECT}/"
+  gcloud_exec_cmds "${GCP_CREDENTIALS_PATH}" "$(pwd)" "gsutil list -p ${DEFAULT_PROJECT} | grep ${TF_STATA_BUCKET_NAME}"
+  if [ -z "${DOCKER_RUN_OUTPUT}" ]; then
+    echo "Terraform backend storage does not exists, creating..."
+    gcloud_exec_cmds "${GOOGLE_APPLICATION_CREDENTIALS_PATH}" "$(pwd)" "gsutil mb -p ${DEFAULT_PROJECT} --pap enforced -b on -l ${DEFAULT_REGION} ${TF_STATA_BUCKET_NAME}"
+  else
+    echo "Terraform backend exists, skip..."
+  fi
+
+  # If the bucket still not exists
+  gcloud_exec_cmds "${GCP_CREDENTIALS_PATH}" "$(pwd)" "gsutil list -p ${DEFAULT_PROJECT} | grep ${TF_STATA_BUCKET_NAME}"
+  if [ -z "${DOCKER_RUN_OUTPUT}" ]; then
+    echo "Unable to creat backend storage bucket, exit..."
+    exit $EXIT_GENERIC_ERR
+  fi
+
+  unset DOCKER_RUN_OUTPUT
+}
+
+destroy_tf_backend() {
+  GCP_CREDENTIALS_PATH="${1}"
+  shift
+  TF_STATA_BUCKET_NAME="gs://tf-state-${DEFAULT_PROJECT}/"
+  gcloud_exec_cmds "${GCP_CREDENTIALS_PATH}" "$(pwd)" "gsutil list -p ${DEFAULT_PROJECT} | grep ${TF_STATA_BUCKET_NAME}"
+  if [ -n "${DOCKER_RUN_OUTPUT}" ]; then
+    echo "Terraform backend storage exists, deleting..."
+    gcloud_exec_cmds "${GOOGLE_APPLICATION_CREDENTIALS_PATH}" "$(pwd)" "gsutil rm -r ${TF_STATA_BUCKET_NAME}"
+  else
+    echo "Terraform backend does not exists, skip..."
+  fi
+  unset DOCKER_RUN_OUTPUT
+  unset TF_STATA_BUCKET_NAME
+}
+
+gcloud_exec_cmds() {
+  GCP_CREDENTIALS_PATH="${1}"
+  shift
+  RUNTIME_SCRIPT_FOLDER="${1}"
+  shift
+  COMMAND_LINE="${1}"
+  shift
+  WORKSPACE_FOLDER="/workspace"
+  # shecllcheck disable=SC2155,SC2046
+  DOCKER_RUN_OUTPUT=$(docker run -it --rm \
+    -e GCP_CREDENTIALS_PATH="${GCP_CREDENTIALS_PATH}" \
+    -v "${RUNTIME_SCRIPT_FOLDER}":"${WORKSPACE_FOLDER}" \
+    -v /etc/localtime:/etc/localtime:ro \
+    -w "${WORKSPACE_FOLDER}" \
+    --volumes-from gcloud-config \
+    --name gcloud_exec_command \
+    "${GCLOUD_CLI_CONTAINER_IMAGE_ID}" sh -c "${COMMAND_LINE}")
+
+  export DOCKER_RUN_OUTPUT
```

Comment:
Instead of this, I suggest checking for the return code.

Reply:
refactored this.